unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+ * @test_procedure Steps:
+ *                      -# Set id_surface and bkgnd_surface_id
+ *                      -# Mocking the surface_get_weston_surface() does return an object
+ *                      -# Calling the surface_event_create()
+ *                      -# Verification point:
+ *                         +# get_id_of_surface() must be called once time
+ *                         +# get_properties_of_surface() must be called once time
+ *                         +# wl_list_init() must be called once time
+ *                         +# surface_get_weston_surface() must be called once time
+ *                         +# wl_list_insert() must be called 2 times
+ *                         +# surface_add_listener() must be called once time
+ *                         +# wl_resource_post_event() must be called {MAX_NUMBER} times
+ *                         +# The result output should same with prepare data
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(ControllerTests, surface_event_create_idDifferentBkgnSurfaceId)
+{
+    struct ivi_layout_surface l_layoutSurface;
+    l_layoutSurface.id_surface = 2;
+    mp_iviShell->bkgnd_surface_id = 1;
+
+    struct weston_surface *lp_westonSurface = (struct weston_surface *)malloc(sizeof(struct weston_surface));
+    SET_RETURN_SEQ(surface_get_weston_surface, &lp_westonSurface, 1);
+
+    surface_event_create(&mp_iviShell->surface_created, &l_layoutSurface);
+
+    ASSERT_EQ(get_id_of_surface_fake.call_count, 1);
+    ASSERT_EQ(get_properties_of_surface_fake.call_count, 1);
+    ASSERT_EQ(wl_list_init_fake.call_count, 1);
+    ASSERT_EQ(surface_get_weston_surface_fake.call_count, 1);
+    ASSERT_EQ(wl_list_insert_fake.call_count, 2);
+    ASSERT_EQ(surface_add_listener_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, MAX_NUMBER);
+    ASSERT_EQ(MAX_NUMBER, g_SurfaceCreatedCount);
+    struct ivisurface *lp_iviSurf = (struct ivisurface*)(uintptr_t(wl_list_init_fake.arg0_history[0]) - offsetof(struct ivisurface, notification_list));
+    ASSERT_EQ(lp_iviSurf->shell, mp_iviShell);
+    ASSERT_EQ(lp_iviSurf->layout_surface, &l_layoutSurface);
+    ASSERT_EQ(mp_iviShell->bkgnd_surface, nullptr);
+
+    free(lp_iviSurf);
+    free(lp_westonSurface);
+}
+
+/** ================================================================================================
+ * @test_id             surface_event_create_idSameBkgnSurfaceId
+ * @brief               Test case of surface_event_create() where id_surface same bkgnd_surface_id
+ * @test_procedure Steps:
+ *                      -# Set id_surface and bkgnd_surface_id
+ *                      -# Mocking the surface_get_weston_surface() does return an object
+ *                      -# Calling the surface_event_create()
+ *                      -# Verification point:
+ *                         +# get_id_of_surface() must be called once time
```

Comment:
I think there are some redundant verification points:

The behaviors of the list (e.g. init, insert, ...) are not required to be checked in this test case (also I see some test cases do the same thing). These behaviors we can separate to one concrete test case for the list.

The only case that needs to be checked is when the behaviors are the result of the sequence in the test case.